src/main/java/com/example/petbuddybackend/service/animal/AnimalService.java

```diff
@@ -60,6 +66,19 @@ public Set<AnimalAttribute> getAnimalAttributesOfAnimal(List<Long> animalAttribu
         return animalAttributeRepository.findDistinctByIdIn(animalAttributeIds);
     }
 
+    public Set<AnimalAttribute> getAnimalAttributes(String animalType, Map<String, List<String>> attributes) {
+        return attributes.entrySet()
+                .stream()
+                .map(entry ->
+                        entry.getValue()
+                                .stream()
+                                .map(value -> getAnimalAttribute(animalType, entry.getKey(), value))
+                                .collect(Collectors.toSet())
+                )
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
+    }
```

Comment:
```suggestion
    public Set<AnimalAttribute> getAnimalAttributes(String animalType, Map<String, List<String>> attributes) {
        return attributes.entrySet().stream()
                .flatMap(entry -> entry.getValue().stream()
                        .map(value -> getAnimalAttribute(animalType, entry.getKey(), value))
                )
                .collect(Collectors.toSet());
    }
```

Jeszcze takie coś znalazłem, że można dać `flatMap` zamiast `map` i troche kod będzie czytelniejszy

Reply:
aaa